lisa/sut_orchestrator/azure/tools.py

```diff
@@ -221,7 +223,7 @@ def _get_waagent_conf_path(self) -> str:
         if result.exit_code == 0:
             waagent_path = result.stdout
         else:
-            if isinstance(self.node.os, CoreOs):
+            if isinstance(self.node.os, CoreOs) and "/usr/share/oem/" in python_cmd:
```

Comment:
Please add it as a sub condition under CoreOs, and add comments in code to explain why the new check condition needs.

Reply:
Thanks for the review @squirrelsc, I've removed this change from the PR since I realized this block will correctly discover the config path in both cases.
```
        result = self.node.execute(
            f'{python_cmd} -c "from azurelinuxagent.common.osutil import get_osutil;'
            'print(get_osutil().agent_conf_file_path)"',
            sudo=use_sudo,
        )
        if result.exit_code == 0:
            waagent_path = result.stdout
```